ballots/tests.py

```diff
@@ -47,6 +46,95 @@ def test_was_published_recently_with_recent_ballot(self):
+        )
+        self.assertEqual(str(ballot), "Test Case Ballot")
+
+    """
+    makes sure the due date of the ballot is after the publish.
+    Don't want a ballot to be past due as soon as it is created.
+    Fixed by clean function in models.py
+    """
+
+
+class QuestionModelTests(TestCase):
+    def test_question_model_str(self):
+        """
+        Making sure the return str for question text is the same str given
+        """
+        ballot = Ballot.objects.create(
+            ballot_title="Test Case Ballot",
+            ballot_description="Making sure the question return str is the same",
+            district="All",
+            pub_date=timezone.now(),
+            due_date=timezone.now() + datetime.timedelta(days=30)
+        )
+        question = Question.objects.create(
+            ballot=ballot,
+            question_text="Make sure the return str is same"
+        )
+        self.assertEqual(str(question), "Make sure the return str is same")
+
+
+class ChoiceModelTests(TestCase):
+    def test_choice_model_str(self):
+        """
+        Making sure the return str for choice text is the same str given
+        """
+        ballot = Ballot.objects.create(
+            ballot_title="Test Case Ballot",
+            ballot_description="Making sure the question return str is the same",
+            district="All",
+            pub_date=timezone.now(),
+            due_date=timezone.now() + datetime.timedelta(days=30)
+        )
+        question = Question.objects.create(
+            ballot=ballot,
+            question_text="Make sure the return str is same"
+        )
+        choice1 = Choice.objects.create(
+            question=question,
+            choice_text="True",
+        )
+        choice2 = Choice.objects.create(
+            question=question,
+            choice_text="False",
+        )
+        self.assertEqual(str(choice1), "True")
+        self.assertEqual(str(choice2), "False")
+
+    def test_choice_vote_count(self):
+        """
+        Making sure the read only votes field can still increment when someone votes
+        """
+        ballot = Ballot.objects.create(
+            ballot_title="Test Case Ballot",
+            ballot_description="Making sure the question return str is the same",
+            district="All",
+            pub_date=timezone.now(),
+            due_date=timezone.now() + datetime.timedelta(days=30)
+        )
+        question = Question.objects.create(
+            ballot=ballot,
+            question_text="Make sure the return str is same"
+        )
+        choice = Choice.objects.create(
+            question=question,
+            choice_text="True",
+            votes=0
+        )
+        self.assertEqual(choice.votes, 0)
+        choice.votes += 1
+        self.assertEqual(choice.votes, 1)
+
     def test_district_matches(self):
         """
         ballot_list should only contain published ballots with districts matching the user's district
```

Comment:
This test fails because ChoiceModelTests does not contain the necessary user and ballot creation in a setup function!

Fixed by adding the following function to ChoiceModelTests:
```python
    def setUp(self):
        self.user = User.objects.create(first_name='John', last_name='Smith', username='testuser',
                                        password='password123', email='JohnSmith@gmail.com')
        self.user.save
        profile = self.user.profile
        profile.ssn = "555-55-5555"
        profile.district = "BaltimoreCounty"
        profile.middle_name = "Jack"
        self.ballot = Ballot.objects.create(ballot_title="Test", district="BaltimoreCounty", pub_date=timezone.now())
```